packages/aws-cdk-lib/aws-codedeploy/lib/host-health-config.ts

```diff
@@ -34,3 +34,42 @@ export class MinimumHealthyHosts {
     return this.json;
   }
 }
+
+/**
+ * Minimum number of healthy hosts per availability zone for a server deployment.
+ */
+export class MinimumHealthyHostsPerZone {
+
+  /**
+   * The minimum healhty hosts threshold expressed as an absolute number.
+   */
+  public static count(value: number): MinimumHealthyHostsPerZone {
+    return new MinimumHealthyHostsPerZone({
+      type: 'HOST_COUNT',
+      value,
+    });
+  }
+
+  /**
+   * The minmum healhty hosts threshold expressed as a percentage of the fleet.
+   */
+  public static percentage(value: number): MinimumHealthyHostsPerZone {
+    return new MinimumHealthyHostsPerZone({
+      type: 'FLEET_PERCENT',
+      value,
+    });
+  }
+
+  private constructor(private readonly json: CfnDeploymentConfig.MinimumHealthyHostsProperty) {
+    if (!Number.isInteger(json.value)) {
+      throw new Error(`The percentage or count value of minimumHealthyHostsPerZone must be an integer, got: ${json.value}`);
+    }
```

Comment:
Is this necessary given the above method declarations `public static count(value: number)` / `public static percentage(value: number)`? 🤔

Reply:
I added validation for percentages just in case users might want to input decimal values, though I expect that users will mostly input whole numbers. If you think it's unnecessary, I can remove it!

Reply:
I agree with keeping the validation, thanks for clarifying 👍

Reply:
Nit: could we also validate that the `value >= 0`?